test/functional/test_framework/test_framework.py

```diff
@@ -542,15 +564,53 @@ def join_network(self):
         connect_nodes(self.nodes[1], 2)
         self.sync_all()
 
-    def sync_blocks(self, nodes=None, **kwargs):
-        sync_blocks(nodes or self.nodes, **kwargs)
-
-    def sync_mempools(self, nodes=None, **kwargs):
-        sync_mempools(nodes or self.nodes, **kwargs)
+    def sync_blocks(self, nodes=None, wait=1, timeout=60):
+        """
+        Wait until everybody has the same tip.
 
-    def sync_all(self, nodes=None, **kwargs):
-        self.sync_blocks(nodes, **kwargs)
-        self.sync_mempools(nodes, **kwargs)
+        sync_blocks needs to be called with an rpc_connections set that has least
```

Comment:
I have a question - it's probably not a big deal to require at least one node to be updated, but would it be appropriate to have an optional `blockhash` parameter and enforce that hash instead?

Reply:
I don't see a use case like that in current code. But I think one benefit your idea will give is to shorten test run time if the block we need to sync to is not usually the tip.